File: algorithms/Strings/Medium/LargestNumber/VSDLargestNumber.java
import java.util.ArrayList;
import java.util.Collections;
import java.util.Comparator;


public class Main{

    static String largestNumber(int[] nums){

        ArrayList<String> arr=new ArrayList<>();

        for(int i:nums) arr.add(String.valueOf(i));
        
        Comparator<String> comp = new Comparator<String>() {
            
            @Override
            public int compare(String str1,String str2){
                return (str2+str1).compareTo(str1+str2);
            }
        };

        Collections.sort(arr,comp);

        StringBuilder ans=new StringBuilder();

        arr.forEach((s)->{
            if(s!="0")
                ans.append(s);
        });
        
        return ans.toString();
    }
    public static void main(String[] args) {

        int[] nums = {3,30,34,5,9};
        
        System.out.println(largestNumber(nums));
    }
}

File: algorithms/Strings/Medium/LargestNumber/VSDLargestNumber.py


import functools as f 


def compare(x,y):   
    return (x<y)-(x>y)


def largestNumber(nums):
    arr=list(map(str,nums))
    arr.sort(key=f.cmp_to_key(lambda x,y:compare(x+y,y+x)))
    return "".join(arr).lstrip("0") or "0"

if __name__=="__main__":
    nums=[3,30,34,5,9]
    print(largestNumber(nums))